packages/dds/matrix/src/matrix.ts

```diff
@@ -57,6 +59,34 @@ interface ISetOpMetadata {
 	localSeq: number;
 	rowsRefSeq: number;
 	colsRefSeq: number;
+	referenceSeqNumber: number;
+}
+
+export interface ISharedMatrixEvents<T> extends ISharedObjectEvents {
+	/**
+	 * Emitted when there is a conflict when this client tries to set a cell value in the {@link SharedMatrix}.
+	 * It could be due to race condition between different clients where this client set op loses or when
+	 * the client tries to set again before its previous set is acked.
+	 * Race Condition: It means we did not know about the Set op from other client when we sent the set op, but
+	 * we received the remote Op and set that value for the cell. Now when we get our local op back, we can not
+	 * overwrite the value due to FWW and need to resolve conflict.
+	 *
+	 * @remarks Listener parameters:
+	 *
+	 * - `currentValue` - The current value of the cell.
+	 *
+	 * - `ignoredValue` - The value that this client tried to set in the cell and got ignored due to conflict.
+	 *
+	 * - `target` - The {@link SharedMatrix} itself.
+	 */
+	(
+		event: "resolveConflict",
+		listener: (
+			currentValue: MatrixItem<T>,
+			ignoredValue: MatrixItem<T>,
+			target: IEventThisPlaceHolder,
```

Comment:
There is no identification to a cell that is being affected, only to matrix instance and values. It will be very hard to deal with such structure.

If we do not have a way to identify cells, then we should probably allow client to pass some kind of cookie when setting value, and pass that cookie back to user. Or we can leverage promises (i.e., result of setting a cell is a promise that resolves when cell content "settles", i.e., we know if it was a loss or winner case). In both cases user of the system can chain continuation and capture enough context to remember cell instance. That said, both cookie and promise solutions have one and the same problem - they are not preservable (in current structure) through container serialization / rehydration. This might result in data loss. Thus, if there is no other choice, I'd rather go with a serializable cookie - something that we could roundtrip through container serialization / rehydration process. Let's chat more about it.

Reply:
Added the row/col number for now.